backend/api/Services/LocalizationService.cs

```diff
@@ -0,0 +1,230 @@
+            {
+                if (ongoingMissionRun is null) { continue; }
+                if (ongoingMissionRun.Status == MissionStatus.Successful) { break; }
+
+                ongoingMissionRun = await _missionRunService.ReadById(ongoingMissionRunId);
+            }
+
+            const string Message = "Timed out while waiting for the localization mission to get an updated status";
+            _logger.LogError("{Message}", Message);
+            throw new TimeoutException(Message);
+        }
+
+        private async Task StartLocalizationMissionInArea(Robot robot, Area missionDefinitionArea)
+        {
+            if (missionDefinitionArea.Deck?.DefaultLocalizationPose?.Pose is null)
+            {
+                const string ErrorMessage = "The mission area is not associated with any deck or that deck does not have a localization pose";
+                _logger.LogError("{Message}", ErrorMessage);
+                throw new DeckNotFoundException(ErrorMessage);
+            }
+            if (robot.Status is not RobotStatus.Available)
+            {
+                string errorMessage = $"Robot '{robot.Id}' is not available as the status is {robot.Status.ToString()}";
+                _logger.LogWarning("{Message}", errorMessage);
+                throw new RobotNotAvailableException(errorMessage);
+            }
+
+            var localizationMissionRun = new MissionRun
+            {
+                Name = "Localization mission",
+                Robot = robot,
+                InstallationCode = missionDefinitionArea.Installation.InstallationCode,
+                Area = missionDefinitionArea,
+                Status = MissionStatus.Pending,
+                DesiredStartTime = DateTimeOffset.UtcNow,
+                Tasks = new List<MissionTask>
+                {
+                    new(missionDefinitionArea.Deck.DefaultLocalizationPose.Pose, "localization")
+                },
+                Map = new MapMetadata()
+            };
+
+            await _missionRunService.Create(localizationMissionRun);
+
+            robot.CurrentArea = localizationMissionRun.Area;
+            await _robotService.Update(robot);
+        }
+
+        private bool RobotIsOnSameDeckAsMission(Area? currentRobotArea, Area? missionArea)
+        {
+            if (currentRobotArea is null)
+            {
+                const string ErrorMessage = "The robot is not associated with an area and a mission may not be started";
+                _logger.LogError("{Message}", ErrorMessage);
+                throw new AreaNotFoundException(ErrorMessage);
+            }
+            if (missionArea is null)
+            {
+                const string ErrorMessage = "The robot is not located on the same deck as the mission as the area has not been set";
+                _logger.LogError("{Message}", ErrorMessage);
+                throw new AreaNotFoundException(ErrorMessage);
+            }
+            if (currentRobotArea?.Deck is null)
```

Comment:
Out of curiosity, do we need this null conditional operator, given that if it is null we already throw an exception?